sympy/solvers/bivariate.py

```diff
@@ -205,11 +321,33 @@ def _solve_lambert(f, symbol, gens):
     if not lamcheck:
         raise NotImplementedError()
 
+    even_degrees = [i for i in _power_list(lhs, symbol).get(symbol, []) \
+                    if i%2 == 0 and i != 0]
+    t = Dummy('t', **symbol.assumptions0)
+
+    # replacing all even_degrees of symbol with dummy variable t.
+    # lhs can also be different from Mul and Add which doesn't need
+    # to be processed.
+    if len(even_degrees) != 0 and (lhs.is_Add or lhs.is_Mul):
+        lhs = lhs.xreplace({symbol**i: t**i for i in even_degrees})
+
+    # check if lhs has the replaced dummy variable t
+    #  and Add for further manipulation.
+    if lhs.is_Add and lhs.has(t):
+        t_term = lhs - lhs.subs(t, 0)
+        rhs = rhs - lhs.subs(t, 0)
+        return _solve_even_degree_expr(expand_log(log(t_term) - log(rhs)), symbol)
+
+    # check if lhs.is_Mul and if it has replaced t variable
+    # then solve it with _solve_even_degree_expr otherwise go further.
     if lhs.is_Mul:
```

Comment:
```suggestion
    if lhs.is_Mul and lhs.has(t):
```

Reply:
```
if lhs.is_Mul:
        lhs = expand_log(log(lhs))
        rhs = log(rhs)
```
@smichr This code has to be run even it has t or not i.e we had to modify lhs and rhs for all type of equation if lhs.is_Mul.

Reply:
ok